### PR TITLE
Experiments with Mask Inputs Probability

### PR DESCRIPTION
**Pull request by anwai98**
(WIP) Experiments without the input mask prompts while iterative prompting in the training scheme

### CODE REVIEW

**Comment by anwai98** [comment]
The complete set of results from mask input probabilities are [here](https://docs.google.com/spreadsheets/d/10GgjA9CDe0y4Ww2hfuma2mBV6L3KmTQLqCKPrTSe9D0/edit?usp=sharing): we see that the batched inference results are a little better when the probability of using mask inputs is less than 1. Let's stick to 0.5 as the default value as we discussed.

**Comment by anwai98** [comment]
I just updated the default value (to `0.5` in the trainer). This now looks good to go. Thanks!

File: micro_sam/training/sam_trainer.py [hunk 1]
```diff
@@ -250,7 +255,14 @@ def _get_updated_points_per_mask_per_subiter(self, masks, sampled_binary_y, batc
 
             _inp["point_coords"] = updated_point_coords
             _inp["point_labels"] = updated_point_labels
-            _inp["mask_inputs"] = logits
+
+            if self.mask_prob > 0:
+                # using mask inputs for iterative prompting while training, with a probability
+                use_mask_inputs = True if random.random() < self.mask_prob else False
```
Comment (constantinpape) [inline micro_sam/training/sam_trainer.py:261]: I am not 100% sure about the order here.

Comment (anwai98) [reply micro_sam/training/sam_trainer.py]: This is now fixed with the latest commit, and is good to go. Thanks!

**Review by constantinpape — APPROVED** [review]
